In [1]:
import pandas as pd
import numpy as np

In [2]:
pd.set_option("display.max_rows", None, "display.max_columns", None)

In [3]:
region_access=pd.read_csv(r"G:\Unidades compartidas\04_PROYECTOS I+D+i\2023 MultiModX\iii) Project\WP3 Scenario definition\Case study input data\Spain\v=0.7\infrastructure\regions_access\regions_access_v0.3.csv")

In [4]:
region_access=region_access.drop(columns="Unnamed: 0")

In [5]:
more_than_one_hour=region_access[(region_access["avg_d2i"]>=60)|(region_access["avg_i2d"]>=60)]

In [6]:
display(more_than_one_hour)

region    station layer pax_type  avg_d2i  avg_i2d
2    ES111       LEVX   air      all      107      105
3    ES112       LEST   air      all       90       91
4    ES113       LEST   air      all      104      105
6    ES114       LEST   air      all       81       82
8    ES120       LEXJ   air      all      136      136
10   ES130       LEBB   air      all       81       79
13   ES212       LEBB   air      all       75       78
17   ES220       LEBB   air      all      134      135
18   ES230       LEBB   air      all      109      108
20   ES241       LEBL   air      all      167      167
23   ES411       LEMD   air      all      134      133
25   ES413       LEMD   air      all      275      274
27   ES415       LEMD   air      all      205      204
28   ES415       LESA   air      all       72       72
29   ES416       LEMD   air      all      112      108
30   ES417       LEMD   air      all      174      175
31   ES418       LEMD   air      all      164      163
33   ES419       LEMD   air      all      224      223
34   ES422       LEMD   air      all      172      175
35   ES423       LEMD   air      all      152      159
36   ES423       LEVC   air      all      202      201
38   ES425       LEMD   air      all       84       85
39   ES431       LEZL   air      all      149      145
40   ES431       LEMD   air      all      285      285
41   ES431       LEBZ   air      all       70       70
42   ES432       LEMD   air      all      231      232
44   ES512       LEBL   air      all       94       91
46   ES513       LEBL   air      all      122      122
48   ES514       LEBL   air      all       76       76
51   ES521       LEVC   air      all      144      144
52   ES522       LEVC   air      all       92       94
58   ES611       LEMG   air      all      184      182
59   ES611       LEAL   air      all      214      217
60   ES612       LEJR   air      all       66       66
61   ES612       LEZL   air      all      119      117
62   ES612       LEMG   air      all      171      174
63   ES613       LEMG   air      all      141      140
64   ES613       LEZL   air      all      119      118
66   ES614       LEMG   air      all      115      114
67   ES615       LEZL   air      all      120      118
68   ES616       LEMG   air      all      181      181
69   ES616       LEMD   air      all      248      249
72   ES620       LEAL   air      all       89       94
82   ES709       GCTS   air      all       74       74
86   ES112  007120300  rail      all       63       64
87   ES112  007131400  rail      all       90       91
89   ES113  007120300  rail      all       61       59
92   ES114  007122100  rail      all       82       81
101  ES220  007181202  rail      all       64       64
104  ES220  007181108  rail      all       67       66
106  ES230  007181110  rail      all       61       60
107  ES230  007181200  rail      all       67       65
110  ES241  007178400  rail      all       62       63
111  ES241  007104040  rail      all       80       79
112  ES241  007170600  rail      all      129      129
113  ES242  007167105  rail      all       89       89
114  ES242  007171201  rail      all      123      123
122  ES412  007111200  rail      all       67       68
125  ES413  007120200  rail      all       76       75
126  ES414  007114100  rail      all       62       62
127  ES414  007111000  rail      all       75       75
130  ES416  007117000  rail      all       91       89
132  ES417  007170403  rail      all       85       86
133  ES417  007170600  rail      all      101      101
134  ES417  007104040  rail      all      149      149
138  ES419  007114100  rail      all      131      130
139  ES419  007110600  rail      all      101      101
140  ES419  007111200  rail      all      229      229
143  ES422  007160400  rail      all       91       91
144  ES422  007137300  rail      all       74       75
145  ES423  007103208  rail      all       77       75
146  ES423  007160600  rail      all      11

These are all the region access that are more than one hour 

In [7]:
more_than_one_hour[more_than_one_hour["layer"]=="air"]

region station layer pax_type  avg_d2i  avg_i2d
2   ES111    LEVX   air      all      107      105
3   ES112    LEST   air      all       90       91
4   ES113    LEST   air      all      104      105
6   ES114    LEST   air      all       81       82
8   ES120    LEXJ   air      all      136      136
10  ES130    LEBB   air      all       81       79
13  ES212    LEBB   air      all       75       78
17  ES220    LEBB   air      all      134      135
18  ES230    LEBB   air      all      109      108
20  ES241    LEBL   air      all      167      167
23  ES411    LEMD   air      all      134      133
25  ES413    LEMD   air      all      275      274
27  ES415    LEMD   air      all      205      204
28  ES415    LESA   air      all       72       72
29  ES416    LEMD   air      all      112      108
30  ES417    LEMD   air      all      174      175
31  ES418    LEMD   air      all      164      163
33  ES419    LEMD   air      all      224      223
34  ES422    LEMD   air      all      172      175
35  ES423    LEMD   air      all      152      159
36  ES423    LEVC   air      all      202      201
38  ES425    LEMD   air      all       84       85
39  ES431    LEZL   air      all      149      145
40  ES431    LEMD   air      all      285      285
41  ES431    LEBZ   air      all       70       70
42  ES432    LEMD   air      all      231      232
44  ES512    LEBL   air      all       94       91
46  ES513    LEBL   air      all      122      122
48  ES514    LEBL   air      all       76       76
51  ES521    LEVC   air      all      144      144
52  ES522    LEVC   air      all       92       94
58  ES611    LEMG   air      all      184      182
59  ES611    LEAL   air      all      214      217
60  ES612    LEJR   air      all       66       66
61  ES612    LEZL   air      all      119      117
62  ES612    LEMG   air      all      171      174
63  ES613    LEMG   air      all      141      140
64  ES613    LEZL   air      all      119      118
66  ES614    LEMG   air      all      115      114
67  ES615    LEZL   air      all      120      118
68  ES616    LEMG   air      all      181      181
69  ES616    LEMD   air      all      248      249
72  ES620    LEAL   air      all       89       94
82  ES709    GCTS   air      all       74       74

airports tend to have a very large catching area, so we will not remove any of them

In [8]:
trains_more_than_one_hour=more_than_one_hour[more_than_one_hour["layer"]=="rail"]

In [9]:
trains_more_than_one_hour.head()

region    station layer pax_type  avg_d2i  avg_i2d
86   ES112  007120300  rail      all       63       64
87   ES112  007131400  rail      all       90       91
89   ES113  007120300  rail      all       61       59
92   ES114  007122100  rail      all       82       81
101  ES220  007181202  rail      all       64       64

we remove the station L'Aldea Amposta 007165402 which is accessible from Jaen, since that does not make a lot of sense

In [10]:
region_access=region_access.drop(index=189)
trains_more_than_one_hour=trains_more_than_one_hour.drop(index=189)

In [11]:
trains_more_than_one_hour=trains_more_than_one_hour.sort_values(["avg_d2i","avg_i2d"],ascending=False)

In [12]:
trains_more_than_one_hour.head()

region    station layer pax_type  avg_d2i  avg_i2d
140  ES419  007111200  rail      all      229      229
177  ES611  007161200  rail      all      152      153
134  ES417  007104040  rail      all      149      149
138  ES419  007114100  rail      all      131      130
112  ES241  007170600  rail      all      129      129

entry 140 is miranda del ebro which is accessible by the nuts of Zamora, which is not really realistic -> I will remove it too

In [13]:
region_access=region_access.drop(index=140)
trains_more_than_one_hour=trains_more_than_one_hour.drop(index=140)

In [14]:
trains_more_than_one_hour.head()

region    station layer pax_type  avg_d2i  avg_i2d
177  ES611  007161200  rail      all      152      153
134  ES417  007104040  rail      all      149      149
138  ES419  007114100  rail      all      131      130
112  ES241  007170600  rail      all      129      129
181  ES613  007151003  rail      all      124      124

entry 177 is station Murcia del Carmen, which is accessible by Almería. I will keep it since Almería is not really well connected.

In [15]:
trains_more_than_one_hour.head()

region    station layer pax_type  avg_d2i  avg_i2d
177  ES611  007161200  rail      all      152      153
134  ES417  007104040  rail      all      149      149
138  ES419  007114100  rail      all      131      130
112  ES241  007170600  rail      all      129      129
181  ES613  007151003  rail      all      124      124

entry 134 is the Zaragoza Delicias station which is accessible via Soria. As these are neighbouring NUTS I will keep it. 

entry 138 is palencia train station which is accessible from Zamora. I will keep it as well.

continue from 112 on Monday
